website/addons/s3/static/s3NodeConfig.js

```diff
@@ -0,0 +1,360 @@
+        self.showSelect(true);
+    }).fail(function(xhr) {
+        var resp = JSON.parse(xhr.responseText);
+        var message = resp.message;
+        var title = resp.title || 'Problem creating bucket';
+        self.creating(false);
+        if (!message) {
+            message = 'Looks like that name is taken. Try another name?';
+        }
+        bootbox.confirm({
+            title: title,
+            message: message,
+            callback: function(result) {
+                if (result) {
+                    self.openCreateBucket();
+                }
+            }
+        });
+    });
+};
+
+ViewModel.prototype.openCreateBucket = function() {
+    var self = this;
+
+    var isValidBucket = /^(?!.*(\.\.|-\.))[^.][a-z0-9\d.-]{2,61}[^.]$/;
+
+    bootbox.prompt('Name your new bucket', function(bucketName) {
+        if (!bucketName) {
+            return;
+        } else if (isValidBucket.exec(bucketName) == null) {
+            bootbox.confirm({
+                title: 'Invalid bucket name',
+                message: 'Sorry, that\'s not a valid bucket name. Try another name?',
+                callback: function(result) {
+                    if (result) {
+                        self.openCreateBucket();
+                    }
+                }
+            });
+        } else {
+            self.createBucket(bucketName);
+        }
+    });
+};
+
+ViewModel.prototype.fetchBucketList = function() {
+    var self = this;
+
+    var ret = $.Deferred();
+    if(self.loadedBucketList()){
+        ret.resolve(self.bucketList());
+    }
+    else{
+         $.ajax({
+            url: self.urls().bucket_list,
+            type: 'GET',
+            dataType: 'json'
+        }).done(function(response) {
+            self.loadedBucketList(true);
+            ret.resolve(response.buckets);
+        })
+        .fail(function(xhr, status, error) {
+            var message = 'Could not retrieve list of S3 buckets at' +
+                'this time. Please refresh the page. If the problem persists, email ' +
+                '<a href="mailto:support@osf.io">support@osf.io</a>.';
+            self.changeMessage(message, 'text-warning');
+            Raven.captureMessage('Could not GET s3 bucket list', {
+                url: self.urls().bucketList,
+                textStatus: status,
+                error: error
+            });
+            ret.reject(xhr, status, error);
+        });
+    }
+    return ret.promise();
+};
+
+ViewModel.prototype.updateFromData = function(settings) {
+    var self = this;
+    self.nodeHasAuth(settings.node_has_auth);
+    self.userHasAuth(settings.user_has_auth);
+    self.userIsOwner(settings.user_is_owner);
+    self.ownerName(settings.owner);
+    self.currentBucket(settings.has_bucket ? settings.bucket : 'None');
+    self.loadedSettings(true);
+    if (settings.urls) {
+        self.urls(settings.urls);
+    }
+};
+
+ViewModel.prototype.fetchFromServer = function(callback) {
```

Comment:
The callback argument should be unnecessary, since you're already returning the jqXHR.